client/src/components/geneExpression/gene.js

```diff
@@ -0,0 +1,191 @@
+// jshint esversion: 6
+/* rc slider https://www.npmjs.com/package/rc-slider */
+
+import React from "react";
+import { connect } from "react-redux";
+
+import { AnchorButton, Icon } from "@blueprintjs/core";
+import Truncate from "../util/truncate";
+import HistogramBrush from "../brushableHistogram";
+
+import * as globals from "../../globals";
+import actions from "../../actions";
+
+@connect((state, ownProps) => {
+  const { gene } = ownProps;
+
+  return {
+    isColorAccessor: state.colors.colorAccessor === gene,
+    isScatterplotXXaccessor: state.controls.scatterplotXXaccessor === gene,
+    isScatterplotYYaccessor: state.controls.scatterplotYYaccessor === gene,
+  };
+})
+class Gene extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      geneIsExpanded: false,
+    };
+  }
+
+  onColorChangeClick = () => {
+    const { dispatch, gene } = this.props;
+    dispatch(actions.requestSingleGeneExpressionCountsForColoringPOST(gene));
+  };
+
+  handleGeneExpandClick = () => {
+    const { geneIsExpanded } = this.state;
+    this.setState({ geneIsExpanded: !geneIsExpanded });
+  };
+
+  handleSetGeneAsScatterplotX = () => {
+    const { dispatch, gene } = this.props;
+    dispatch({
+      type: "set scatterplot x",
+      data: gene,
+    });
+  };
+
+  handleSetGeneAsScatterplotY = () => {
+    const { dispatch, gene } = this.props;
+    dispatch({
+      type: "set scatterplot y",
+      data: gene,
+    });
+  };
+
+  handleDeleteGeneFromSet = () => {
+    const { dispatch, gene, geneset } = this.props;
+
+    dispatch({
+      type: "geneset: delete genes",
+      genesetName: geneset,
+      geneSymbols: [gene],
+    });
+  };
+
+  render() {
+    const {
+      gene,
+      isColorAccessor,
+      isScatterplotXXaccessor,
+      isScatterplotYYaccessor,
+      isDiffexp,
+    } = this.props;
+    const { geneIsExpanded } = this.state;
+    return (
+      <div>
+        <div
+          style={{
+            marginLeft: 5,
+            marginRight: 0,
+            marginTop: 2,
+            display: "flex",
+            justifyContent: "space-between",
+            alignItems: "center",
+          }}
+        >
+          <span
+            role="menuitem"
+            tabIndex="0"
+            data-testclass="gene-expand"
+            data-testid={`${gene}:gene-expand`}
+            onKeyPress={/* todo_genesets */ () => {}}
```

Comment:
is this TODO captured in an issue, or should it be in this PR?

Reply:
It's for accessibility so that users can tab. Here's the jslint error if it is removed. I will capture as an issue.

```
 104:11  error  Visible, non-interactive elements with click handlers must have at least one keyboard listener  jsx-a11y/click-events-have-key-events

✖ 1 problem (1 error, 0 warnings)
```

Reply:
https://github.com/chanzuckerberg/cellxgene/issues/2101

Reply:
I like this a lot as a convention

Reply:
Most of our keyboard commands do work. I'm going to keep this issue open to figure this out.